src/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -228,10 +228,7 @@ public void SendFile_APM(IPAddress listenAt, bool sendPreAndPostBuffers, int byt
                 });
 
                 // Wait for the tasks to complete
-                Task<Task> firstCompleted = Task.WhenAny(serverTask, clientTask);
-                Assert.True(firstCompleted.Wait(TestTimeout), "Neither client nor server task completed within allowed time");
-                firstCompleted.Result.GetAwaiter().GetResult();
-                Assert.True(Task.WaitAll(new[] { serverTask, clientTask }, TestTimeout), $"Tasks didn't complete within allowed time. Server:{serverTask.Status} Client:{clientTask.Status}");
+                Assert.True(await (new[] { serverTask, clientTask }).WhenAllWithTimeout(TestTimeout), $"Tasks didn't complete within allowed time. Server:{serverTask.Status} Client:{clientTask.Status}");
```

Comment:
While many of these should be the WhenAllCompletedOrAnyFailed helper, this one in particular should be.  The previous code was doing that, just with blocking.